connect/src/com/telenor/connect/headerenrichment/HeLogic.java

```diff
@@ -24,6 +24,7 @@
     private static final long HE_TOKEN_TIMEOUT_MILLISECONDS = 10_000; // Timeout is a best guess
     // at what would be long enough to succeed and short enough for users to wait it out if
     // it should fail/time out, and do a normal flow
+    private static boolean heWasExecutedAndReceivedSuccessResponse = false;
```

Comment:
When will this ever get back to `false`? When the app is closed and restarted? So it might survive for instance a SIM change?

Reply:
The logic had changed. Please, check the new version out.